controlplane/kubeadm/controllers/fakes_test.go

```diff
@@ -57,20 +55,6 @@ func (f *fakeManagementCluster) GetMachinesForCluster(c context.Context, n clien
 	return f.Machines, nil
 }
 
-func (f *fakeManagementCluster) TargetClusterControlPlaneIsHealthy(_ context.Context, _ client.ObjectKey) error {
-	if !f.ControlPlaneHealthy {
```

Comment:
It seems we are losing the possibility of controlling the result of `ControlPlaneHealthy` func during the refactor.
If instead, this is a leftover not actually used in tests, we should remove  `ControlPlaneHealthy` field from `fakeManagementCluster`

Reply:
This is no longer in use as scale up test was using this before and we moved `reconcileHealth` above.
All the other tests are setting `EtcdHealthy` to true. So no need to add this if we don't plan to add more tests.